registration/registration-services/src/main/java/io/mosip/registration/mdm/service/impl/MosipDeviceSpecificationFactory.java

```diff
@@ -213,15 +214,139 @@ private void initByPort(Integer availablePort) {
 						addToDeviceInfoMap(getKey(deviceType, deviceSubType), bioDevice);
 					}
 				}
+				ApplicationContext.setApplicationMap(unSupportedVersionsMap);
+				ApplicationContext.setApplicationMap(unavailableVersionsMap);
+				deviceSize = mdmBioDevices.size();
 			}
 		} catch (RuntimeException runtimeException) {
 			LOGGER.error("Failed to parse / validate MDM response", runtimeException);
 		}
 	}
 
+	private List<MdmBioDevice> getMdmBioDevices(List<MdmBioDevice> mdmBioDevices) {
+		LOGGER.info("Entering getMdmBioDevices method with mdmBioDevices: " + mdmBioDevices);
+		List<MdmBioDevice> devices = new ArrayList<>();
+		ObjectMapper objectMapper = new ObjectMapper();
+
+		if (mdmBioDevices != null) {
+			for (MdmBioDevice device : mdmBioDevices) {
+				LOGGER.info("Processing device: " + device);
+				boolean isAvailable = true;
+				boolean isVersionAvailable = true;
+				try {
+					if (device.getDeviceProviderName() != null && device.getDeviceType() != null) {
+						String propertyValue = null;
+						switch (device.getDeviceType().toLowerCase()) {
+							case "face":
+								propertyValue = ApplicationContext.getStringValueFromApplicationMap(RegistrationConstants.SBI_UNSUPPORTED_FACE);
+								break;
+							case "finger":
+								propertyValue = ApplicationContext.getStringValueFromApplicationMap(RegistrationConstants.SBI_UNSUPPORTED_FINGER);
+								break;
+							case "iris":
+								propertyValue = ApplicationContext.getStringValueFromApplicationMap(RegistrationConstants.SBI_UNSUPPORTED_IRIS);
+								break;
+							default:
+								LOGGER.warn("Unknown device type: " + device.getDeviceType());
+						}
+						LOGGER.debug("Device type: " + device.getDeviceType() + ", Property value: " + propertyValue);
+
+						if (propertyValue != null) {
+							Map<String, Object> propertyMap = objectMapper.readValue(propertyValue, HashMap.class);
+							Map<String, List<String>> versions = new HashMap<>();
+							LOGGER.info("Cast the values from Object to List<String>");
+							for (Map.Entry<String, Object> entry : propertyMap.entrySet()) {
+								Object value = entry.getValue();
+								if (value instanceof List) {
+									versions.put(entry.getKey(), (List<String>) value);
+								}
+							}
+							LOGGER.debug("Versions map: " + versions);
+
+							String providerName = device.getDeviceProviderName().toUpperCase();
+							if (versions.containsKey(providerName)) {
+								List<String> unsupportedVersions = versions.get(providerName);
+								boolean isVersionUnSupported = unsupportedVersions.stream()
+										.anyMatch(unsupportedVersion -> compareVersions(device.getSerialVersion(), unsupportedVersion));
```

Comment:
@Manas0606 Is the SBI version stored here: device.getSerialVersion()? The SBI version is stored in the MdmDeviceInfo, serviceVersion variable

Reply:
yes sbi version stores in device.getSerialVersion()